### PR TITLE
Fix Cataclysm Transporters

### PR DESCRIPTION
**Pull request by aaron02**
**Description**
In cata and beyond is no Movementflag Transport

Fix: #1100 

**Todo / Checklist**
- [x] Target is branch develop.
- [x] Detailed description about this pull request.
- [x] Checked AE-Coding standards.

**Tests Performed:** 
- [x] Build AE with "TREAT_WARNINGS_AS_ERRORS" flag turned on.
- [x] Server startup.
- [x] Log into world.

**Multiversion Ingame Tests Performed:**
- [x] Cata

### CODE REVIEW

File: src/world/Movement/MovementGenerators/IdleMovementGenerator.cpp [hunk 1]
```diff
@@ -152,6 +172,16 @@ void DistractMovementGenerator::initialize(Unit* owner)
...
     init.Launch();
 }
+#else
+void DistractMovementGenerator::initialize(Unit* owner)
+{
+    // Distracted creatures stand up if not standing
+    if (!owner->getStandState())
+        owner->setStandState(STANDSTATE_STAND);
+
+    owner->addUnitStateFlag(UNIT_STATE_DISTRACTED);
```
Comment (Appled) [inline src/world/Movement/MovementGenerators/IdleMovementGenerator.cpp:182]: Is this state flag needed in classic - wotlk?

**Comment by schnek** [comment]
yes it solves the problem 😉 only the non-pch build also fails, open a new ticket?

**Comment by schnek** [comment]
yes i only checked "cata"

i check with .npc spawn 28312 and i have problems with Vehicle (no movement).

**Comment by schnek** [comment]
up

File: src/world/Movement/MovementGenerators/IdleMovementGenerator.cpp (hunk 1 above)
Comment (aaron02) [reply src/world/Movement/MovementGenerators/IdleMovementGenerator.cpp:182]: Some Lines Above its in the BaseUnitState, and there we launch a MoveSplineInit and this uses the initialized State

**Comment by aaron02** [comment]
sql syntax must ahve been there before and is not a part of this pr. so maybe make an issue about it,
vehicle also is not affected by this pr also make an issue for this problem ^^
maybe in the future i find some spare time to look at it

**Comment by schnek** [comment]
ok, I'll try not to forget to open the "issue" 
view non-pch build ☺

**Review by Appled — APPROVED** [review]

**Pull request merged**